enterprise/server/ociregistry/ociregistry.go

```diff
@@ -173,6 +173,83 @@ func (r *registry) handleV2Request(ctx context.Context, w http.ResponseWriter, i
 	}
 }
 
+func makeUpstreamRequest(ctx context.Context, method, acceptHeader, authorizationHeader string, ociResourceType ocipb.OCIResourceType, ref gcrname.Reference) (*http.Response, error) {
+	var path string
+	switch ociResourceType {
+	case ocipb.OCIResourceType_BLOB:
+		path = "/v2/" + ref.Context().RepositoryStr() + "/blobs/" + ref.Identifier()
+	case ocipb.OCIResourceType_MANIFEST:
+		path = "/v2/" + ref.Context().RepositoryStr() + "/manifests/" + ref.Identifier()
+	case ocipb.OCIResourceType_UNKNOWN:
+		return nil, fmt.Errorf("unknown OCI resource type, expected blobs or manifests")
+	}
+	u := url.URL{
+		Scheme: ref.Context().Scheme(),
+		Host:   ref.Context().RegistryStr(),
+		Path:   path,
+	}
+	upreq, err := http.NewRequest(method, u.String(), nil)
+	if err != nil {
+		return nil, fmt.Errorf("could not make %s request to upstream registry '%s': %s", method, u.String(), err)
+	}
+	if acceptHeader != "" {
+		upreq.Header.Set(headerAccept, acceptHeader)
+	}
+	if authorizationHeader != "" {
+		upreq.Header.Set(headerAuthorization, authorizationHeader)
+	}
+	upresp, err := http.DefaultClient.Do(upreq.WithContext(ctx))
```

Comment:
you can remove the intermediate variables and directly `return http.DefaultClient.Do(...)`